### PR TITLE
Made gRPC instrumenter compliant with semantic conventions

### LINKED ISSUE #201: Make gRPC instrumenter compliant with semantic conventions
The gRPC instrumenter needs and overhaul to be compliant with the Opentelemetry semantic conventions.

Please assign this to me.

### PR DESCRIPTION
**Pull request by prydin**
* Changed span name to comply with conventions

* Added ```rpc.service``` span attribute

* Added ```net.peer.*``` attributes to client and server span.

Closes #201

### CODE REVIEW

File: instrumentation/grpc-1.5/src/main/java/io/opentelemetry/auto/instrumentation/grpc/server/TracingServerInterceptor.java [hunk 1]
```diff
@@ -57,7 +62,11 @@ private TracingServerInterceptor() {}
       spanBuilder.setNoParent();
     }
     final Span span = spanBuilder.startSpan();
-    span.setAttribute(MoreTags.RESOURCE_NAME, call.getMethodDescriptor().getFullMethodName());
+    span.setAttribute(MoreTags.RESOURCE_NAME, methodName);
```
Comment (trask) [inline instrumentation/grpc-1.5/src/main/java/io/opentelemetry/auto/instrumentation/grpc/server/TracingServerInterceptor.java]:
```suggestion
```
same comment as above

File: instrumentation/grpc-1.5/src/test/groovy/GrpcStreamingTest.groovy [hunk 2]
```diff
@@ -68,9 +69,17 @@ class GrpcStreamingTest extends AgentTestRunner {
...
-    ManagedChannel channel = InProcessChannelBuilder.forName(getClass().name).build()
+    def port = PortUtils.randomOpenPort()
+    Server server = ServerBuilder.forPort(port).addService(greeter).build().start()
+    ManagedChannelBuilder channelBuilder = ManagedChannelBuilder.forAddress("localhost", port)
+
+    // Depending on the version of gRPC usePlainText may or may not take an argument.
+    try {
+      channelBuilder.usePlaintext()
+    } catch (MissingMethodException e) {
+      channelBuilder.usePlaintext(true)
```
Comment (trask) [inline instrumentation/grpc-1.5/src/test/groovy/GrpcStreamingTest.groovy]: Nice, dealing with different versions is so clean with groovy! 😆

File: instrumentation/grpc-1.5/src/test/groovy/GrpcStreamingTest.groovy [hunk 3]
```diff
@@ -102,14 +111,18 @@ class GrpcStreamingTest extends AgentTestRunner {
     assertTraces(1) {
       trace(0, 2) {
         span(0) {
-          operationName "grpc.client"
+          operationName "example.Greeter/Conversation"
           spanKind CLIENT
           parent()
           errored false
           tags {
             "$MoreTags.RESOURCE_NAME" "example.Greeter/Conversation"
```
Comment (trask) [inline instrumentation/grpc-1.5/src/test/groovy/GrpcStreamingTest.groovy]:
```suggestion
```
etc

File: instrumentation/grpc-1.5/src/main/java/io/opentelemetry/auto/instrumentation/grpc/client/GrpcClientBuilderInstrumentation.java [hunk 4]
```diff
@@ -38,7 +43,8 @@ public GrpcClientBuilderInstrumentation() {
 
   @Override
   public ElementMatcher<TypeDescription> typeMatcher() {
-    return named("io.grpc.internal.AbstractManagedChannelImplBuilder");
+    return safeHasSuperType(named("io.grpc.ManagedChannelBuilder"))
+        .or(named("io.grpc.ManagedChannelBuilder"));
```
[the remark this answers is not in the document]
Comment (prydin) [reply instrumentation/grpc-1.5/src/main/java/io/opentelemetry/auto/instrumentation/grpc/client/GrpcClientBuilderInstrumentation.java]: Done!

File: instrumentation/grpc-1.5/src/main/java/io/opentelemetry/auto/instrumentation/grpc/common/GrpcHelper.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+import io.grpc.ManagedChannelBuilder;
+import io.opentelemetry.auto.bootstrap.WeakMap;
+import io.opentelemetry.auto.instrumentation.api.MoreTags;
+import io.opentelemetry.trace.Span;
+import java.net.InetSocketAddress;
+
+public class GrpcHelper {
+  private static final WeakMap<ManagedChannelBuilder, InetSocketAddress> builderToAddress =
+      WeakMap.Provider.newWeakMap();
```
[the remark this answers is not in the document]
Comment (prydin) [reply instrumentation/grpc-1.5/src/main/java/io/opentelemetry/auto/instrumentation/grpc/common/GrpcHelper.java]: Done!

File: instrumentation/grpc-1.5/src/main/java/io/opentelemetry/auto/instrumentation/grpc/client/TracingClientInterceptor.java [hunk 6]
```diff
@@ -30,27 +30,34 @@
...
   public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
       final MethodDescriptor<ReqT, RespT> method,
       final CallOptions callOptions,
       final Channel next) {
 
-    final Span span = TRACER.spanBuilder("grpc.client").setSpanKind(CLIENT).startSpan();
-    span.setAttribute(MoreTags.RESOURCE_NAME, method.getFullMethodName());
+    final String methodName = method.getFullMethodName();
+    final Span span = TRACER.spanBuilder(methodName).setSpanKind(CLIENT).startSpan();
+    span.setAttribute(MoreTags.RESOURCE_NAME, methodName);
```
[the remark this answers is not in the document]
Comment (prydin) [reply instrumentation/grpc-1.5/src/main/java/io/opentelemetry/auto/instrumentation/grpc/client/TracingClientInterceptor.java]: Done!

File: instrumentation/grpc-1.5/src/main/java/io/opentelemetry/auto/instrumentation/grpc/server/TracingServerInterceptor.java (hunk 1 above)
Comment (prydin) [reply instrumentation/grpc-1.5/src/main/java/io/opentelemetry/auto/instrumentation/grpc/server/TracingServerInterceptor.java]: Done

File: instrumentation/grpc-1.5/src/test/groovy/GrpcStreamingTest.groovy (hunk 3 above)
Comment (prydin) [reply instrumentation/grpc-1.5/src/test/groovy/GrpcStreamingTest.groovy]: Done

**Pull request merged**